From afe3c32e490dd1af1e9ad8233154f262815f168a Mon Sep 17 00:00:00 2001
From: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
Date: Wed, 3 Apr 2024 14:13:44 +0100
Subject: [PATCH 1/2] cryptlib_mbedtls: use external Mbed TLS

Do not include mbedtls headers files from libspdm fork of Mbed TLS.
This change is RMM specific, for building cryptlib_mbedtls library
from RMM.

Signed-off-by: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
---
 os_stub/cryptlib_mbedtls/CMakeLists.txt       | 6 +-----
 os_stub/cryptlib_mbedtls/internal_crypt_lib.h | 4 ++--
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/os_stub/cryptlib_mbedtls/CMakeLists.txt b/os_stub/cryptlib_mbedtls/CMakeLists.txt
index ba00b51142..fd3cdd9396 100644
--- a/os_stub/cryptlib_mbedtls/CMakeLists.txt
+++ b/os_stub/cryptlib_mbedtls/CMakeLists.txt
@@ -9,10 +9,6 @@ INCLUDE_DIRECTORIES(${LIBSPDM_DIR}/include
                     ${LIBSPDM_DIR}/os_stub/include
                     ${LIBSPDM_DIR}/os_stub
                     ${LIBSPDM_DIR}/os_stub/cryptlib_mbedtls
-                    ${LIBSPDM_DIR}/os_stub/mbedtlslib/include
-                    ${LIBSPDM_DIR}/os_stub/mbedtlslib/include/mbedtls
-                    ${LIBSPDM_DIR}/os_stub/mbedtlslib/mbedtls/include
-                    ${LIBSPDM_DIR}/os_stub/mbedtlslib/mbedtls/include/mbedtls
 )
 
 SET(src_cryptlib_mbedtls
@@ -47,4 +43,4 @@ if(CMAKE_SYSTEM_NAME MATCHES "Windows")
     if((TOOLCHAIN STREQUAL "CBMC") OR (TOOLCHAIN STREQUAL "VS2015") OR (TOOLCHAIN STREQUAL "VS2019") OR (TOOLCHAIN STREQUAL "VS2022"))
         TARGET_COMPILE_OPTIONS(cryptlib_mbedtls PRIVATE /wd4090)
     endif()
-endif()
\ No newline at end of file
+endif()
diff --git a/os_stub/cryptlib_mbedtls/internal_crypt_lib.h b/os_stub/cryptlib_mbedtls/internal_crypt_lib.h
index 5fd6035f44..a32cc0dd5d 100644
--- a/os_stub/cryptlib_mbedtls/internal_crypt_lib.h
+++ b/os_stub/cryptlib_mbedtls/internal_crypt_lib.h
@@ -18,10 +18,10 @@
 #include "library/cryptlib.h"
 #include "spdm_crypt_ext_lib/cryptlib_ext.h"
 
-/* We should alwasy add mbedtls/config.h here
+/* We should alwasy add mbedtls_config.h here
  * to ensure the config override takes effect.*/
 
-#include <mbedtls/config.h>
+#include <mbedtls_config.h>
 
 int libspdm_myrand(void *rng_state, unsigned char *output, size_t len);
 
-- 
2.39.2

